exercises/darts/canonical-data.json

```diff
@@ -25,52 +25,88 @@
     },
     {
       "property": "score",
-      "description": "A dart lands in the outer circle",
+      "description": "On middle circle",
       "input": {
-        "x": 4,
-        "y": 4
+        "x": -5,
+        "y": 0
       },
-      "expected": 1
+      "expected": 5
     },
     {
       "property": "score",
-      "description": "A dart lands right in the border between outer and middle circles",
+      "description": "On inner circle",
       "input": {
-        "x": 5,
-        "y": 0
+        "x": 0,
+        "y": -1
       },
-      "expected": 5
+      "expected": 10
     },
     {
       "property": "score",
-      "description": "A dart lands in the middle circle",
+      "description": "Near centre",
       "input": {
-        "x": 0.8,
-        "y": -0.8
+        "x": -0.1,
+        "y": -0.1
       },
-      "expected": 5
+      "expected": 10
     },
     {
       "property": "score",
-      "description": "A dart lands right in the border between middle and inner circles",
+      "description": "Just within inner circle",
       "input": {
-        "x": 0,
-        "y": -1
+        "x": 0.7,
+        "y": 0.7
       },
       "expected": 10
     },
     {
       "property": "score",
-      "description": "A dart lands in the inner circle",
+      "description": "Just beyond inner circle",
       "input": {
-        "x": -0.1,
-        "y": -0.1
+        "x": 0.8,
+        "y": -0.8
       },
-      "expected": 10
+      "expected": 5
     },
     {
       "property": "score",
-      "description": "A dart whose coordinates sum to > 1 but whose radius to origin is <= 1 is scored in the inner circle",
+      "description": "Just within middle circle",
+      "input": {
+        "x": -3.5,
+        "y": 3.5
+      },
+      "expected": 5
+    },   
+    {
+      "property": "score",
+      "description": "Just beyond middle circle",
+      "input": {
+        "x": -3.6,
+        "y": -3.6
+      },
+      "expected": 1
+    },     
+    {
+      "property": "score",
+      "description": "Just within outer circle",
+      "input": {
+        "x": -7.0,
+        "y": 7.0
+      },
+      "expected": 1
+    },     
+    {
+      "property": "score",
+      "description": "Just beyond outer circle",
+      "input": {
+        "x": 7.1,
+        "y": -7.1
+      },
+      "expected": 0
+    },    
+    {
+      "property": "score",
+      "description": "Asymmetric position within inner circle",
```

Comment:
that is interesting. I see that asymmetry (not being on the lines `x = y` or `x = -y`) is coincidentally a trait of the cases added in https://github.com/exercism/problem-specifications/pull/1488, but I don't think it is necessary for the thing being tested, with the following example: x = 3.5, y = 3.5

* is on `x = y`
* sums to > 5
* radius is <= 5

Therefore, asymmetry is not the actual thing being tested in these test cases, and writing their description as asymmetric loses information as to why this test case is here. Some possible choices I see here are:

* Decide that losing this information from the description is okay, with the justification being that it is possible to obtain it by reviewing commit history.
* Find some way to write this in the description that is still acceptable. Since I am raising this point it behooves me to give some suggestions... unfortunately some of my suggestions require some form of jargon
    * Within inner circle by Euclidean distance, not Manhattan
    * Within inner circle by Euclidean distance, not taxicab
    * Within inner circle by radius, not addition
    * Within inner circle but not its inscribed square
* Maybe this is a situation where nesting these three cases is appropriate, under a `cases` whose description describes what's in common between these three test cases.

Reply:
I liked all four suggestions. My preference would be the third as the least exotic, closely followed by the first as well defined terms.  

However, thanks @petertseng for the background on the purpose of those #1488 tests.  The existing description **"A dart whose coordinates sum to > 1 but whose radius to origin is <= 1 is scored in the inner circle"** didn't impart to me that it was to prevent **"an erroneous-but-passing solution where the sum of the absolute values of the coordinates of the dart are compared to the radius thresholds"**. This is a good example where the "what" didn't impart the "why" to me, and your third suggestion better describes "why".

Now considering the three test pairs I arranged [near either side of each boundary, e.g...](https://github.com/exercism/problem-specifications/blob/2f71f0332fe0f96008842fc00158b74d3cc333eb/exercises/darts/canonical-data.json#L72-L86)
```
    {
      "property": "score",
      "description": "Just within middle circle",
      "input": {
        "x": -3.5,
        "y": 3.5
      },
      "expected": 5
    },   
    {
      "property": "score",
      "description": "Just beyond middle circle",
      "input": {
        "x": -3.6,
        "y": -3.6
      },
      "expected": 1
    },     
```
simple addition won't be able to satisfy all three pairs.   

So it seems I inadvertently made those #1488 tests redundant.  
They might either be removed or re-purposed for asymmetry as I've done.

Reply:
I checked and you are right that the "just within" tests all will detect an implementation that tests absolute addition:

```
== anti-verify abs score ==
5. Just within inner circle: Got 5 instead of 10
7. Just within middle circle: Got 1 instead of 5
9. Just within outer circle: Got 0 instead of 1
11. Asymmetric position within inner circle: Got 5 instead of 10
12. Asymmetric position between inner and middle circles: Got 1 instead of 5
13. Asymmetric position between middle and outer circles: Got 0 instead of 1
```

So you are correct that these asymmetric points are all redundant.

I would say the answer for "should these tests be kept?" would depend on whether it is useful (would find any mistaken implementations) to have any points that aren't on `x = y` and `x = -y`. I wasn't sure if there are any, so I can be convinced either way. Though note that there are already some earlier test cases that show that.

I think it perhaps makes the most sense to remove these three tests then? Should the other three tests' descriptions be changed from "just within" to something else resembling the above suggestions? Or is "just within" enough information to figure it out?

I think I'm good trusting your judgment either way at this point

Reply:
I can't imagine anyone really doing it, but theoretically without an asymmetric test students might  
```
get away with... sqrt(2*x*x) 
rather than... sqrt(x*x + y*y) 
```
But one asymmetric test should do. Culled the other two.

Reply:
@petertseng Does the updated canonical data now match your suggestions?

Reply:
I feel I have said what I wanted to say and that I do not object to the current situation. I remind others that I think it might still be possible to incorporate "by radius not by addition" (or whatever choice is chosen) if it will actually help a student understand what is wrong with a submitted implementation. if "just outside" (or whatever) is deemed to be sufficient, I accept the collective judgment.